doc/compiler/il/SymbolsSymrefsAliasing.md

```diff
@@ -268,3 +268,67 @@ It does a reachability analysis from all catch blocks and mark all local variabl
 symbol references that can be used. If there is a store before a load for the same 
 symbol reference, it will not mark the symbol reference for the load. Shadows and 
 statics are not considered in the catchLocalUse bit vector calculation.
+
+## `findOrCreateShadowSymbol` vs `findOrFabricateShadowSymbol`
+
+`findOrCreateShadowSymbol` and `findOrFabricateShadowSymbol` are two
+functions provided to construct shadow symbols. The key distinction between
+them is as follows.
+
+`findOrCreateShadowSymbol` is intended for creating shadow symbols that
+directly correspond to an entity in the source code. For example, any regular
+field access on an object is represented by a shadow constructed by this
+function.
+
+`findOrFabricateShadowSymbol` is intended for creating shadow symbols
+that have *no* corresponding entity in the source code. Having a different
+API for such cases has the advantage of providing the opportunity to optimize
+the created symbols. Specifically, very liberal assumptions can be made about
+the aliasing of the symref associated with the new symbol. For example, since
+the symbol does not appear in the source code, it may be safe to assume
+that it cannot alias things that do appear in the code. Hence, 
+`findOrFabricateShadowSymbol` allows aliasing of Generic Int Shadows to
+be suppressed.
+
+Because both of these functions are able to "find" symbols created/fabricated
+by the other, there are some important subtleties that arise from using both
+to access the shame symbol. The compiler relies on certain properties that
```

Comment:
shame = same

Reply:
It really is a shame :stuck_out_tongue: 

Fixed.